src/game/puppyprint.c

```diff
@@ -671,17 +687,81 @@ void puppyprint_profiler_process(void) {
         if (sDebugOption >= (sizeof(ppPages) / sizeof(struct PuppyPrintPage))) {
             sDebugOption = 0;
         }
+    } else {
+        if (sPPDebugPage == PUPPYPRINT_PAGE_LEVEL_SELECT)
+        {
+            if (gPlayer1Controller->buttonPressed & U_JPAD) {
+                sLevelSelectOption--;
+                // If there is no level entry to this ID, skip over.
+                while (sLevelNames[sLevelSelectOption][0] == 0 && sLevelSelectOption < LEVEL_COUNT) {
+                    sLevelSelectOption--;
+                }
+                if (sLevelSelectOption <= 0) {
+                    sLevelSelectOption = LEVEL_COUNT - 1;
+                    // If there is no level entry to this ID, skip over.
+                    while (sLevelNames[sLevelSelectOption][0] == 0 && sLevelSelectOption < LEVEL_COUNT) {
+                        sLevelSelectOption--;
+                    }
+                }
+            }
+            if (gPlayer1Controller->buttonPressed & D_JPAD) {
+                sLevelSelectOption = (sLevelSelectOption + 1) % (LEVEL_COUNT - 1);
+                // If there is no level entry to this ID, skip over.
+                while (sLevelNames[sLevelSelectOption][0] == 0 && sLevelSelectOption < LEVEL_COUNT) {
+                    sLevelSelectOption = (sLevelSelectOption + 1) % (LEVEL_COUNT - 1);
+                }
+            }
+            if (gPlayer1Controller->buttonPressed & R_JPAD) {
+                sLevelSelectOptionArea++;
+                if (sLevelSelectOptionArea > AREA_COUNT - 1) {
+                    sLevelSelectOptionArea = 0;
+                }
+            }
+            if (gPlayer1Controller->buttonPressed & L_JPAD) {
```

Comment:
This is a counterintuitive control, might be worth just remapping to A considering the game is paused at this point anyway. Might as well add B to cancel it or something as well.

Reply:
Maybe if that seems a bit too weird, you could remap it to use joystick instead of or in addition to D-Pad also. Maybe D-Pad  could have exclusive control over warp node selection? Just speculating here, idk how intuitive execution would be in practice.